src/PackageVersions/FallbackVersions.php

```diff
@@ -64,20 +64,27 @@ private static function getPackageData() : array
             __DIR__ . '/../../composer.lock',
         ];
 
+        $packageData = [];
         foreach ($checkedPaths as $path) {
             if (file_exists($path)) {
+                $data = json_decode(file_get_contents($path), true);
                 switch (basename($path)) {
                     case 'installed.json':
-                        return json_decode(file_get_contents($path), true);
+                        $packageData[] = $data;
+                        break;
                     case 'composer.lock':
-                        $data = json_decode(file_get_contents($path), true);
-                        return array_merge($data['packages'], $data['packages-dev'] ?? []);
+                        $packageData[] = $data['packages'] + ($data['packages-dev'] ?? []);
+                        break;
                     default:
                         // intentionally left blank
                 }
             }
         }
 
+        if (!empty($packageData)) {
```

Comment:
Please use `!== []`: the `empty()` construct is inherently dangerous and should never be used

Reply:
Hm. I haven't heard of that before. Do you have a reference as to why not to use empty?

Reply:
The most basic issue is that `empty($undefinedVar)` still works, but besides that, we know the input data structure, and should assert precisely on it, not on an empty-ish concept (`empty()` operates on various different types)

Reply:
Ok, I see. Yes, I already knew those reasons. It's a habit I use especially for user input. :)